Generals/Code/GameEngine/Source/GameLogic/Object/Behavior/PropagandaTowerBehavior.cpp

```diff
@@ -205,11 +206,28 @@ UpdateSleepTime PropagandaTowerBehavior::update( void )
 		}
 	}
 
-	if( self->getContainedBy()  &&  self->getContainedBy()->getContainedBy() )
+#if RETAIL_COMPATIBLE_CRC
+	Bool contained = self->getContainedBy() && self->getContainedBy()->getContainedBy();
+#else
+	Object* container = self->getContainedBy();
+	Bool contained = false;
+
+	while (container)
+	{
+		ContainModuleInterface* containModule = container->getContain();
+		if (containModule && containModule->isEnclosingContainerFor(self))
```

Comment:
This seems odd in the loop because `isEnclosingContainerFor(self)` would only ever be true for the direct parent container, no?